### PR TITLE
[LPT] Precision restriction customization extending

### PR DESCRIPTION
**Pull request by eshoguli**
### Details:
 - *Precision restriction customization extending + usage for 3D convolution in CPU*
 - *Tests: https://github.com/openvinotoolkit/openvino/pull/17196*

### Tickets:
 - *104566*

### CODE REVIEW

File: src/common/low_precision_transformations/src/markup_precisions.cpp [hunk 1]
```diff
@@ -113,13 +117,21 @@ bool ngraph::pass::low_precision::MarkupPrecisions::run_on_model(const std::shar
...
                 }
 
-                const pass::low_precision::PrecisionsRestriction::PrecisionsByPorts& precisionsByPorts = it2->second;
-                setRestriction(node, precisionsByPorts);
+                const Restriction::RestrictionByVersion& precisionsByPorts = it2->second;
+                setRestriction(
+                    node,
+                    (precisionsByPorts.precisionsFunction != nullptr) ?
+                        precisionsByPorts.precisionsFunction(node) :
+                        precisionsByPorts.precisions);
```
Comment (v-Golubev) [inline src/common/low_precision_transformations/src/markup_precisions.cpp:125]: Can we encapsulate this precisions computation logic in `setRestriction` method or in some `Restriction::RestrictionByVersion` method?

File: src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp [hunk 2]
```diff
@@ -478,9 +479,19 @@ void Transformations::Lpt(const bool hasINT16orINT32Levels, const std::vector<ov
                     {{0}, {ov::element::u8, ov::element::i8}},
                     {{1}, {ov::element::i8}}
                 }),
-            PrecisionsRestriction::create<ov::opset1::GroupConvolution>({
+            PrecisionsRestriction::create<ov::opset1::GroupConvolution>([input0LowPrecisionList](const std::shared_ptr<ngraph::Node>& node){
+                const auto& input_partial_shape = node->get_input_partial_shape(0);
+                const auto& rank = input_partial_shape.rank();
+                if (rank.is_static() && (rank.get_length() == 5ull)) {
```
Comment (v-Golubev) [inline src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp:485]: Minor correction
```suggestion
            PrecisionsRestriction::create<ov::opset1::GroupConvolution>([&input0LowPrecisionList](const std::shared_ptr<ov::Node>& node){
                const auto& input_partial_shape = node->get_input_partial_shape(0);
                const auto rank = input_partial_shape.rank();
                if (rank.is_static() && (rank.get_length() == 5)) {
```

File: src/common/low_precision_transformations/include/low_precision/markup_precisions.hpp [hunk 3]
```diff
@@ -38,13 +38,26 @@ class ngraph::pass::low_precision::MarkupPrecisions : public ngraph::pass::Funct
...
     public:
+        class RestrictionByVersion {
+        public:
+            RestrictionByVersion() = default;
+            RestrictionByVersion(
+                const std::function<PrecisionsRestriction::PrecisionsByPorts(const std::shared_ptr<Node>&)>& precisionsFunction,
+                const PrecisionsRestriction::PrecisionsByPorts& precisions) :
+                precisionsFunction(precisionsFunction),
+                precisions(precisions) {}
+
```
Comment (v-Golubev) [inline src/common/low_precision_transformations/include/low_precision/markup_precisions.hpp:49]: Can we introduce a method that will return `PrecisionsRestriction::PrecisionsByPorts` for a node depending on class fields?

File: src/common/low_precision_transformations/include/low_precision/common/precisions_restriction.hpp [hunk 4]
```diff
@@ -41,23 +41,39 @@ class PrecisionsRestriction {
     ngraph::Node::type_info_t operationType;
     bool specifyVersion;
     PrecisionsByPorts precisionsByPorts;
+    std::function<PrecisionsByPorts(const std::shared_ptr<Node>&)> precisionsByPortsFunction;
```
[the remark this answers is not in the document]
Comment (eshoguli) [reply src/common/low_precision_transformations/include/low_precision/common/precisions_restriction.hpp:44]: done, thanks

File: src/common/low_precision_transformations/src/markup_precisions.cpp (hunk 1 above)
Comment (eshoguli) [reply src/common/low_precision_transformations/src/markup_precisions.cpp:125]: done, thanks

File: src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp (hunk 2 above)
Comment (eshoguli) [reply src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp:485]: done

File: src/common/low_precision_transformations/include/low_precision/markup_precisions.hpp (hunk 3 above)
Comment (eshoguli) [reply src/common/low_precision_transformations/include/low_precision/markup_precisions.hpp:49]: done: `get`

**Review by v-Golubev — APPROVED** [review]

**Review by sshlyapn — APPROVED** [review]

**Pull request merged**